src/main/scala/com/sergigp/quasar/event/AsyncEventBus.scala

```diff
@@ -3,37 +3,33 @@ package com.sergigp.quasar.event
 import scala.concurrent.{ExecutionContext, Future}
 import scala.reflect.ClassTag
 
-import org.slf4j.Logger
-
 final class AsyncEventBus(
-  logger: Logger,
   onSuccess: String => Unit = _ => (),
-  onFailure: Throwable => Unit = _ => (),
-  onHandlerNotFound: String => Unit = _ => (),
+  onFailure: Throwable => Unit = _ => ()
 )(implicit ec: ExecutionContext)
     extends EventBus[Future] {
 
-  private var handlers = Map.empty[Class[_], Event => Future[Unit]]
+  private var handlers: Map[Class[_], List[Event => Future[Unit]]] = Map.empty
 
-  override def publish[E <: Event](event: E): Future[Unit] =
-    handlers
-      .get(event.getClass) match {
-      case Some(handler) => handleEvent(event, handler)
-      case None          =>
-        onHandlerNotFound(event.getClass.getSimpleName)
-        Future.unit
+  override def publish[E <: Event](event: E): Future[Unit] = {
+    val results = handlers.getOrElse(event.getClass, List.empty[Event => Future[Unit]]).map { handler =>
+      handleEvent(event, handler)
+    }
+    Future.sequence(results).map(_ => ()).recoverWith {
+      case _ => Future.successful(())
     }
+  }
 
   override def subscribe[E <: Event: ClassTag](handler: E => Future[Unit]): Unit = {
```

Comment:
Could this be injected instead of mutating every time?

Reply:
what will happen if a handler needs the eventbus as a dependency? :)

Reply:
If it's a singleton, nothing 😬✌️

Reply:
(or lazy 💤)